index.rst

```diff
@@ -1,61 +1,263 @@
+Summary
+=======
 
-.. sectnum::
+After considerable effort fine tunning the DDS middleware configuration, we were finally able to obtain a stable system, that is capable of operating at large scale with low middleware-related failure rate.
+At the current advanced state of the project, which is approaching its final construction stages, one might be tempted to accept this part of the project as concluded.
 
-.. TODO: Delete the note below before merging new content to the main branch.
+Nevertheless, as we demonstrated, there are a number of issues hiding underneath that may pose significant problems in the future, or even be seen as violating system requirements.
 
-.. note::
+Overall our experience with DDS has been, to say the least, underwhelming.
+Even thought the technology is capable of achieving impressive throughput and latency, in reality, it proved to be extremely cumbersome and hard to manage and debug on large scale systems.
+On top of if all we also face a potentially end-of-life cycle of the adopted library, which makes the problem considerably worse.
 
-   **This technote is not yet published.**
+After extensive exploring different alternatives to the problem we propose a potential long term solution to the problem, namely, replace DDS by the already in-use Kafka.
+Our benchmarks shows that Kafka is able to fulfill our system throughput and latency requirements.
+We also shown that transitioning to Kafka would require minimum effort and minimum code refactor.
 
-   This technote gathers information about the history, current state and contemplates future evolutions of the Vera Rubin Observatory Control System (Rubin-OCS) Middleware. The middleware is the backbone of the Rubin-OCS. The highly distributed nature of the Rubin-OCS places tight constraints in terms of latency, availability and reliability for the middleware. Here we gather information to answer some common questions regarding technology choices, describe some of the in-house work done to obtain a stable system and document some of the concerns with the current state of the system, potential impacts for near-future/commissioning and future/operations. We also cover some of the work we have been doing to investigate alternative technologies and suggest some potential roadmaps for the future of the system.
+We also note that there are major advantages of transitioning to Kafka before the end of construction.
+To begin with, we take advantage of a "standing army", as developers are actively engaged with the system and motivated.
+Furthermore, it also gives us the opportunity to perform the transition in a time when uptime pressure is not as large as it will become once commissioning of the main telescope commences.
+Given our development cycle and the current state of the system we expect to be able to fully transition to Kafka in a 1 to 2 deployment cycles (1-3 months approximately), with no impact to the summit and minimum to no downtime on the Tucson Test Stand.
```

Comment:
Nice!

Reply:
Really?
Sorry for being a doubting thomas, but we rarely ever make our estimates.  I would like to see how this estimate was scoped out, especially if I have to defend it.  Given the size and breath of this effort (and all the detail/exception cases being vaguely glossed over - as the last 10% always takes 90% of the effort) and the amount of time it took to configure DDS.....
I see this as happening over 1 to 2 years and finishing as we start into operations
I know we want to sell this thing to make sure it goes through, but we don't want to blow up in our faces.